src/clients/packages.client.ts

```diff
@@ -0,0 +1,51 @@
+import {Input} from "../types"
+import { Octokit } from "@octokit/core"
+import { throttling } from "@octokit/plugin-throttling"
+import { getOctokit } from "@actions/github"
+import { GitHub } from "@actions/github/lib/utils"
+import { RequestOptions } from "@octokit/types"
+import { info, warning } from "@actions/core"
+
+export class PackagesClient {
```

Comment:
I like this much better than the abstract class. I think we can further simplify by making this a function instead, since the class is only used to assign the `client` and get it back then. Something like:

```ts
export function packagesClient(input: Input): InstanceType<typeof GitHub> {
  ...
}
```

Reply:
Agreed - updated